ehcache-core/src/test/java/org/ehcache/core/PrefixLoggerTest.java

```diff
@@ -0,0 +1,140 @@
+package org.ehcache.core;
+
+import junit.framework.TestCase;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+public class PrefixLoggerTest extends TestCase {
+
+  @Test
+  @SuppressWarnings("try")
+  public void testSimplePrefix() {
+
+    try(EhcachePrefixLoggerFactory.Context ignored = EhcachePrefixLoggerFactory.withContext("cache-alias", "cache-name")){
+      Logger logger = EhcachePrefixLoggerFactory.getLogger(PrefixLoggerTest.class);
+      logger.info("Message {}", "Message");
+    }
+
+  }
+
+  @Test @SuppressWarnings("try")
+  public void testEscapedPrefix() {
+
+    try(EhcachePrefixLoggerFactory.Context ignored = EhcachePrefixLoggerFactory.withContext("cache-alias", "cache{}name")){
+      Logger logger = EhcachePrefixLoggerFactory.getLogger(PrefixLoggerTest.class);
+      logger.info("Message {}", "Message");
+    }
+
+  }
+
+  @Test @SuppressWarnings("try")
+  public void testDoubleEscapedPrefix() {
+
+    try(EhcachePrefixLoggerFactory.Context ignored = EhcachePrefixLoggerFactory.withContext("cache-alias", "cache\\{}name")){
+      Logger logger = EhcachePrefixLoggerFactory.getLogger(PrefixLoggerTest.class);
+      logger.info("Message {}", "Message");
+    }
+
+  }
+
+  @Test @SuppressWarnings("try")
+  public void testTripleEscapedPrefix() {
+
+    try(EhcachePrefixLoggerFactory.Context ignored = EhcachePrefixLoggerFactory.withContext("cache-alias", "cache\\\\{}name")){
+      Logger logger = EhcachePrefixLoggerFactory.getLogger(PrefixLoggerTest.class);
+      logger.info("Message {}", "Message");
+    }
+  }
+
+  @Test
+  public void testAllMethodsDelegationWithoutEnrichment() throws InvocationTargetException, IllegalAccessException {
```

Comment:
I don't think you need both the enriched and unenriched versions of this test. I would just delete this one in favor of the improved version.

Reply:
ok, I have removed this.